sdk/js/src/index.ts

```diff
@@ -627,9 +627,39 @@ export class Client {
         },
         program: "limo",
       };
+    } else if (opportunity.program === "swap") {
+      let tokens;
+      if (opportunity.tokens.type === "input_token_specified") {
+        tokens = {
+          type: "input_specified",
+          inputToken: {
+            amount: BigInt(opportunity.tokens.input_token.amount),
+            token: new PublicKey(opportunity.tokens.input_token.token),
+          },
+          outputToken: new PublicKey(opportunity.tokens.output_token),
+        } as const;
+      } else {
+        tokens = {
+          type: "output_specified",
+          inputToken: new PublicKey(opportunity.tokens.input_token),
+          outputToken: {
+            amount: BigInt(opportunity.tokens.output_token.amount),
+            token: new PublicKey(opportunity.tokens.output_token.token),
+          },
+        } as const;
+      }
+      return {
+        chainId: opportunity.chain_id,
+        slot: opportunity.slot,
+        opportunityId: opportunity.opportunity_id,
+        program: "swap",
+        permissionAccount: new PublicKey(opportunity.permission_account),
+        routerAccount: new PublicKey(opportunity.router_account),
+        userWalletAddress: new PublicKey(opportunity.user_wallet_address),
+        tokens,
+      };
     } else {
-      console.warn("Cannot handle wallet opportunities");
-      return undefined;
+      console.warn("Unsupported opportunity", opportunity);
```

Comment:
we have `constructSvmBid` in the client, it may be good to have `constructSvmSwap` as part of the client for consistency